components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/dao/ClaimDAO.java

```diff
@@ -173,4 +173,80 @@ public int getClaimId(Connection connection, String claimDialectURI, String clai
 
         return claimId;
     }
+
+    public Map<String, String> getClaimProperties(Connection connection, int claimId, int
+            tenantId) throws ClaimMetadataException {
+
+        Map<String, String> claimProperties = new HashMap<>();
+
+        PreparedStatement prepStmt = null;
+        ResultSet rs = null;
+
+        String query = SQLConstants.GET_CLAIM_PROPERTIES;
+
+        try {
+            prepStmt = connection.prepareStatement(query);
+            prepStmt.setInt(1, claimId);
+            prepStmt.setInt(2, tenantId);
+            rs = prepStmt.executeQuery();
+
+            while (rs.next()) {
+                String claimPropertyName = rs.getString(SQLConstants.PROPERTY_NAME_COLUMN);
+                String claimPropertyValue = rs.getString(SQLConstants.PROPERTY_VALUE_COLUMN);
+
+                claimProperties.put(claimPropertyName, claimPropertyValue);
+            }
```

Comment:
Need to commit the connection.
To close connection, prepared statement and result set better use try with resource.
Do this for all the newly added DAO methods.